src/test/regress/sql/multi_schema_support.sql

```diff
@@ -721,9 +721,111 @@ SET citus.task_executor_type TO "adaptive";
 
 
 -- test ALTER TABLE SET SCHEMA
--- we expect that it will warn out
 SET search_path TO public;
-ALTER TABLE test_schema_support.nation_hash SET SCHEMA public;
+
+CREATE SCHEMA old_schema;
+CREATE TABLE old_schema.table_set_schema(id int);
+SELECT create_distributed_table('old_schema.table_set_schema', 'id');
+CREATE SCHEMA new_schema;
+
+SELECT objid::oid::regnamespace as "Distributed Schemas"
+    FROM citus.pg_dist_object
+    WHERE objid::oid::regnamespace IN ('old_schema', 'new_schema');
+\c - - - :worker_1_port
+SELECT table_schema AS "Shards' Schema"
+    FROM information_schema.tables
+    WHERE table_name LIKE 'table\_set\_schema\_%' AND
+          table_schema IN ('old_schema', 'new_schema', 'public')
+    GROUP BY table_schema;
+\c - - - :master_port
+
+ALTER TABLE old_schema.table_set_schema SET SCHEMA new_schema;
+
+SELECT objid::oid::regnamespace as "Distributed Schemas"
+    FROM citus.pg_dist_object
+    WHERE objid::oid::regnamespace IN ('old_schema', 'new_schema');
+\c - - - :worker_1_port
+SELECT table_schema AS "Shards' Schema"
+    FROM information_schema.tables
+    WHERE table_name LIKE 'table\_set\_schema\_%' AND
+          table_schema IN ('old_schema', 'new_schema', 'public')
+    GROUP BY table_schema;
+\c - - - :master_port
+
+DROP SCHEMA old_schema CASCADE;
+DROP SCHEMA new_schema CASCADE;
+
+
+-- test ALTER TABLE SET SCHEMA from public
+CREATE TABLE table_set_schema(id int);
+SELECT create_distributed_table('table_set_schema', 'id');
+CREATE SCHEMA new_schema;
+
+SELECT objid::oid::regnamespace as "Distributed Schemas"
+    FROM citus.pg_dist_object
+    WHERE objid='new_schema'::regnamespace::oid;
+\c - - - :worker_1_port
+SELECT table_schema AS "Shards' Schema"
+    FROM information_schema.tables
+    WHERE table_name LIKE 'table\_set\_schema\_%' AND
+          table_schema IN ('new_schema', 'public')
+    GROUP BY table_schema;
+\c - - - :master_port
+
+ALTER TABLE table_set_schema SET SCHEMA new_schema;
```

Comment:
maybe add a query after this to make sure it still works.

Reply:
@halilozanakgul was this implemented? It looks like the query on line 788 does this. Could we mark this resolved if that is the case?